Core/Src/segment.c

```diff
@@ -59,457 +106,556 @@ const int32_t VOLT_TEMP_CALIB_OFFSET = 0;
+	adBms6830_Adcv(RD_ON, CONTINUOUS, DCP_OFF, RSTF_OFF, OW_OFF_ALL_CH);
+	adBmsPollAdc(PLCADC);
+
+	adbms_wake();
+	read_adbms_data(chips, RDCVALL, Rdcvall, ALL_GRP);
+	// read_adbms_data(chip, RDCVA, Cell, A);
+	// read_adbms_data(chip, RDCVB, Cell, B);
+	// read_adbms_data(chip, RDCVC, Cell, C);
+	// read_adbms_data(chip, RDCVD, Cell, D);
+	// read_adbms_data(chip, RDCVE, Cell, E);
+	// read_adbms_data(chip, RDCVF, Cell, F);
+}
+
+/**
+ * @brief Get voltages from the S-ADCs.
+ * 
+ * @param chip Array of chips to get voltage readings from.
+ */
+void get_s_adc_voltages(cell_asic chips[NUM_CHIPS])
+{
+	write_config_regs(chips);
+	adbms_wake();
+	adBms6830_Adsv(CONTINUOUS, DCP_OFF, OW_OFF_ALL_CH);
+	adBmsPollAdc(PLSADC);
+
+	adbms_wake();
+	read_adbms_data(chips, RDSALL, Rdsall, ALL_GRP);
+	// read_adbms_data(chip, RDSVA, S_volt, A);
+	// read_adbms_data(chip, RDSVB, S_volt, B);
+	// read_adbms_data(chip, RDSVC, S_volt, C);
+	// read_adbms_data(chip, RDSVD, S_volt, D);
+	// read_adbms_data(chip, RDSVE, S_volt, E);
+	// read_adbms_data(chip, RDSVF, S_volt, F);
+}
+
+/**
+ * @brief Get the avgeraged cell voltages.
+ * 
+ * @param chip Array of chips to get voltage readings of.
+ */
+void get_avgd_cell_voltages(cell_asic chips[NUM_CHIPS])
+{
+	write_config_regs(chips);
+	adbms_wake();
+	adBms6830_Adcv(RD_ON, CONTINUOUS, DCP_OFF, RSTF_OFF, OW_OFF_ALL_CH);
+	adBmsPollAdc(PLCADC);
+
+	adbms_wake();
+	read_adbms_data(chips, RDACALL, Rdacall, ALL_GRP);
+}
+
+/**
+ * @brief Get the filtered cell volrages.
+ * 
+ * @param chip Array of chips to get voltage readings of.
+ */
+void get_filtered_cell_volrages(cell_asic chips[NUM_CHIPS])
+{
+	write_config_regs(chips);
+	adbms_wake();
+	adBms6830_Adcv(RD_ON, CONTINUOUS, DCP_OFF, RSTF_OFF, OW_OFF_ALL_CH);
+	adBmsPollAdc(PLCADC);
+
+	adbms_wake();
+	read_adbms_data(chips, RDFCALL, Rdfcall, ALL_GRP);
+}
+
+/**
+ * @brief Get the c and s adc voltages. Does this with RDCSALL command.
+ * 
+ * @param chips Array of chips to get voltage readings of.
+ */
+void get_c_and_s_adc_voltages(cell_asic chips[NUM_CHIPS])
+{
+	write_config_regs(chips);
+	adbms_wake();
+	adBms6830_Adcv(RD_ON, CONTINUOUS, DCP_OFF, RSTF_OFF, OW_OFF_ALL_CH);
+	adBmsPollAdc(PLCADC);
+
+	adbms_wake();
+	read_adbms_data(chips, RDCSALL, Rdcsall, ALL_GRP);
+}
+
+/**
+ * @brief Read every register connected to the AUX ADC.
+ * 
+ * @param chips Array of chips to get voltage readings of.
+ */
+void read_aux_registers(cell_asic chips[NUM_CHIPS])
+{
+	write_config_regs(chips);
+	adBms6830_Adax(AUX_OW_OFF, PUP_DOWN, AUX_ALL);
+	adBmsPollAdc(PLAUX1);
+
+	adbms_wake();
+	read_adbms_data(chips, RDAUXA, Aux, A);
+	read_adbms_data(chips, RDAUXB, Aux, B);
+	read_adbms_data(chips, RDAUXC, Aux, C);
+	read_adbms_data(chips, RDAUXD, Aux, D);
+}
+
+/**
+ * @brief Read voltages in every register connected to AUX2 ADC.
+ * 
+ * @param chips Array of chips to get voltages of.
+ */
+void read_aux2_registers(cell_asic chips[NUM_CHIPS])
+{
+	write_config_regs(chips);
```

Comment:
for all read_* commands.  Why do we write to registers each time?  Feels like we dont need to?

Reply:
If the chip slept, all the config registers are wiped (I think).